ibis/backends/bigquery/tests/unit/test_client.py

```diff
@@ -30,3 +31,23 @@ def test_parse_project_and_dataset_raises_error():
     expected_message = "data-project.my_dataset.table is not a BigQuery dataset"
     with pytest.raises(ValueError, match=expected_message):
         client.parse_project_and_dataset("my-project", "data-project.my_dataset.table")
+
+
+@pytest.mark.parametrize(
+    "bq_path_str",
+    [
+        "ibis-gbq.ibis_gbq_testing.argle",
+        "ibis-gbq.ibis_gbq_testing.28argle",
+        "mytable-287a",
+        "myproject.mydataset.my-table",
+        "my-dataset.mytable",
+        "a-7b0a.dev_test_dataset.test_ibis5",
+    ],
+)
+def test_force_quoting(bq_path_str):
+    table = sg.parse_one(bq_path_str, into=sg.exp.Table, read="bigquery")
+    table = _force_quote_table(table)
+
+    for key in ("this", "db", "catalog"):
+        if (val := table.args[key]) is not None:
+            assert val.quoted
```

Comment:
can we instead test the expected output as part of the test parameterization?

Reply:
yep